src/typechecker/FStar.TypeChecker.Rel.fst

```diff
@@ -891,8 +937,8 @@ let destruct_flex_t' t : flex_t =
     | _ -> failwith "Not a flex-uvar"
 
 (* Destruct a term into its uvar head and arguments *)
-let destruct_flex_t env t wl : flex_t * worklist =
-  let t, wl = ensure_no_uvar_subst env t wl in
+let destruct_flex_t t wl : flex_t * worklist =
+  let t, wl = ensure_no_uvar_subst wl.tcenv t wl in
```

Comment:
Should this be (p_env wl)? Or should we just remove the env argument of ensure_no_uvar_subst?

Reply:
Ah, we don't have a prob here, so can't call p_env.
But, I wonder then if ensure_no_uvar_subst's use of the wl.tcenv is safe

Reply:
It's safe as it only uses the env to check for debug flags. Added some comments. Same for may_relate below (for delta_depths in that case)